File: src/main/java/com/yepngo/qrgen/renderers/pixel/CirclesRenderer.java
package com.yepngo.qrgen.renderers.pixel;

import com.yepngo.qrgen.renderers.common.ImgParameters;

import java.awt.*;

public class CirclesRenderer extends IndependentPixelRenderer {
    public CirclesRenderer(ImgParameters imgParams) {
        super(imgParams);
    }

    @Override
    protected void drawActualShape(ImgParameters imgParams, Graphics2D gfx) {
        gfx.fillOval(0, 0, imgParams.getCellSize(), imgParams.getCellSize());
    }
}

File: src/main/kotlin/com/yepngo/qrgen/renderers/pixel/CirclesRenderer.kt
package com.yepngo.qrgen.renderers.pixel

import com.yepngo.qrgen.renderers.common.ImgParameters
import java.awt.Graphics2D

internal class CirclesRenderer(
    imgParams: ImgParameters,
) : IndependentPixelRenderer(imgParams) {
    override fun drawActualShape(
        imgParams: ImgParameters,
        gfx: Graphics2D,
    ) {
        gfx.fillOval(0, 0, imgParams.cellSize, imgParams.cellSize)
    }
}
